elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchRel.java

```diff
@@ -64,6 +65,15 @@
      */
     final List<String> groupBy = new ArrayList<>();
 
+    /**
+     * Keeps mapping between calcite expression identifier (like {@code EXPR$0}) and
+     * original item call like {@code _MAP['foo.bar']} ({@code foo.bar} really).
+     * This information otherwise might be lost during query translation.
+     *
+     * @see SqlStdOperatorTable#ITEM
+     */
+    final List<Map.Entry<String, String>> expressionItemMap = new ArrayList<>();
```

Comment:
Possibly, you need just a list with the field names in the right order and not a mapping of the complete expression.